pins/drivers.py

```diff
@@ -15,14 +16,38 @@
 
 
 def _assert_is_pandas_df(x):
-    import pandas as pd
+    df_family = _get_df_family(x)
 
-    if not isinstance(x, pd.DataFrame):
+    if df_family != "pandas":
         raise NotImplementedError(
             "Currently only pandas.DataFrame can be saved to a CSV."
         )
 
 
+def _get_df_family(df) -> Literal["unknown", "pandas", "polars"]:
+    try:
+        import polars as pl
+    except ModuleNotFoundError:
+        is_polars_df = False
+    else:
+        is_polars_df = isinstance(df, pl.DataFrame)
+
+    import pandas as pd
+
+    is_pandas_df = isinstance(df, pd.DataFrame)
+
+    if not is_polars_df and not is_pandas_df:
+        return "unknown"
+    if is_polars_df and is_pandas_df:  # Hybrid DataFrame type!
+        return "unknown"
+    elif is_polars_df:
+        return "polars"
+    elif is_pandas_df:
+        return "pandas"
+    else:
+        assert_never(df)
```

Comment:
OR, WDYT of raising `NotImplementedError` here rather than returning "unknown", since we already know "unknown" will cause a failure? (We could `try/except NotImplementedError` for creating the default title potentially)

Reply:
I like the `NotImplementedError` idea a lot 😄